qrand/platforms/qsharp/circuit.py

```diff
@@ -28,76 +28,129 @@
 ###############################################################################
 class QsharpCircuit(QuantumCircuit):
     def __init__(self, num_qubits: int) -> None:
-        self.ERROR_MSG = f"{self.__class__.__name__}"  # TODO
-        raise NotImplementedError(self.ERROR_MSG)
+        self.gates: str = ""
+        self._num_qubits: int = num_qubits
+
+    def _validate_qubit_index(self, qubit_index: int) -> None:
+        if qubit_index >= self.num_qubits:
+            raise ValueError(
+                f"Qubit index out of range {qubit_index} >= {self.num_qubits}."
+            )
 
     @property
     def num_qubits(self) -> int:
-        raise NotImplementedError(self.ERROR_MSG)
+        return self._num_qubits
 
     ############################## SPECIAL GATES ##############################
     def measure(self, target_qubit: int) -> None:
-        raise NotImplementedError(self.ERROR_MSG)
+        self._validate_qubit_index(target_qubit)
+        self.gates += f"""
+        if M(q[{target_qubit}]) == One{{
+        set value w/= {target_qubit} <- "1";
+        }}
```

Comment:
This might be a bit easier to read, Q# isn't whitespace sensitive.
```suggestion
        if M(q[{target_qubit}]) == One 
        {{
            set value w/= {target_qubit} <- "1";
        }}
```

Reply:
Got it.